src/Tasks/Microsoft.NET.Build.Tasks/LockFileExtensions.cs

```diff
@@ -97,7 +97,46 @@ public static LockFileTargetLibrary GetLibrary(this LockFileTarget lockFileTarge
 
             return projectDeps;
         }
-        
+
+        public static HashSet<string> GetProjectFileDependencySet(this LockFile lockFile)
+        {
+            string GetPackageNameFromDependency(string dependency)
```

Comment:
I'd suggest a comment here explaining what the format of `dependency` is and thus why we're looking for the whitespace.